dna/__main__.py

```diff
@@ -290,6 +292,122 @@ def evaluate_handler(
         if not os.listdir(plot_dir):
             os.rmdir(plot_dir)
 
+def configure_tuning_parser(parser):
+    parser.add_argument(
+        '--model', type=str, action='store', required=True,
+        help='the model name to tune'
+    )
+    parser.add_argument(
+        '--model-config-path', type=str, action='store', required=True,
+        help='the directory to read in the original model config'
+    )
+    parser.add_argument(
+        '--tuning-config-path', type=str, action='store', required=True,
+        help='the directory to read in the tuning config'
+    )
+    parser.add_argument(
+        '--train-path', type=str, action='store', required=True,
+        help='path to read the train data'
+    )
+    parser.add_argument(
+        '--test-path', type=str, action='store', default=None,
+        help='path to read the test data; if not provided, train data will be split'
+    )
+    parser.add_argument(
+        '--test-size', type=int, action='store', default=1,
+        help='the number of datasets in the test split'
+    )
+    parser.add_argument(
+        '--split-seed', type=int, action='store', default=0,
+        help='seed used to split the data into train and test sets'
+    )
+    parser.add_argument(
+        '--k', type=int, action='store', default=10,
+        help='the number of pipelines to rank'
+    )
+    parser.add_argument(
+        '--model-seed', type=int, default=1,
+        help='seed used to control the random state of the model'
+    )
+    parser.add_argument(
+        '--output-dir', type=str, default=None,
+        help='directory path to write outputs for this model run'
+    )
+    parser.add_argument(
+        '--metafeature-subset', type=str, default='all', choices=['all', 'landmarkers', 'non-landmarkers']
+    )
+    parser.add_argument(
+        '--verbose', default=False, action='store_true'
+    )
+
+def tuning_handler(arguments_global: argparse.Namespace, data_resolver=get_data):
+    # gather config files
+    with open(arguments_global.model_config_path, "r") as file:
+        model_config = json.load(file)
+    with open(arguments_global.tuning_config_path, "r") as file:
+        tuning_config = json.load(file)
+
+    # define function to evaluate fitness
+    def _evaluate_model(model_config):
```

Comment:
this function seems like a lot of duplicated code. can we refactor existing code so we don't have to duplicate it?

Reply:
I agree - I just don't know how to do that with argparse.  How do you combine two?

Reply:
we may have to factor out a common function that takes the same inputs and outputs everything that is needed for both use cases. let's look at it when you come in

Reply:
It was easier than I thought.  Done.

Reply:
I am not seeing the changes...

Reply:
It's not showing on github for some reason.  Look at the actual code?  I can see the changes

Reply:
I looked at the commits in this PR and the branch in the main part of the repo, but I don't see changes from the last time I looked at this. I see that `_evaulate_mode` looks a lot like `evaluate_handler`. I am not sure what you are seeing.

Reply:
Can you see my other thread of comments?  If you still can't see it on github check it out.  I can pull the latest code and see the changes

Reply:
Or perhaps we're miscommunicating on what is similar. Which functions?  I was talking about the `handlers` for tuning/evaluate

Reply:
If you mean `_evaluate_mode` and the regular `evaluate` function, they have to be seperate.  One will take in different parameters and return different values (has to return a tuple with a score for tuningdeap while the input is only the config)

Reply:
I appreciate that they may have different inputs/outputs. The core functionality is largely the same. That should be factored out so we don't have duplicate code.

Reply:
I’m not sure I see an easy way to do that - if you see one feel free to do it.  The evaluate function has a lot of other code in it that makes it hard to abstract further.  Otherwise I’ll look at it next Wednesday or Thuraday